tracks/x86-64-assembly/exercises/two-fer/mentoring.md

```diff
@@ -0,0 +1,110 @@
+### Reasonable solutions
+
+#### Hand-rolled loop
+
+```nasm
+    movzx edx, byte [rdi]
+    test dl, dl
+    je .loop_end
+.loop_start:
+    add rdi, 1
+    mov byte [rsi], dl
+    add rsi, 1
+    movzx edx, byte [rdi]
+    test dl, dl
+    jne .loop_start
+.loop_end:
+```
+
+#### Using `rep movsb` instructions
+
+```nasm
+    mov rdi, rsi
+    lea rsi, [rel symbol]
+    mov rcx, symbol_len
+    rep movsb
+```
+
+### Common errors
+
+- Forgetting to save registers.
+
+    > Registers `%rbp`, `%rbx` and `%r12` through `%r15` “belong” to the calling function and the called function is required to preserve their values. In other words, a called function must preserve these registers’ values for its caller. (Matz, Hubička, Jaeger, & Mitchell, 2014, p. 15)
```

Comment:
I am not sure how to get to page 15 of that document... but I did find "AMD64 ABI Draft 0.99 – December 7, 2007" on page 15.

Reply:
Found the footnote, just hadn't gotten that far down the page yet. ;)

Reply:
I hit approve, but also look forward to seeing the suggested formatting changes come in as well.

Reply:
Did I miss something? I thought I made all the suggested changes. Which formatting changes are we talking about?

Reply:
Perhaps the conversations just weren't marked as resolved.

Reply:
https://github.com/exercism/website-copy/pull/1352#discussion_r336793646 is an example.

Reply:
Yeah they show as outdated, just not resolved!